src/rgw/rgw_op.cc

```diff
@@ -908,6 +909,25 @@ void rgw_build_iam_environment(rgw::sal::Driver* driver,
   }
 }
 
+void handle_replication_status_header(
+    const DoutPrefixProvider *dpp,
+    rgw::sal::Attrs& attrs,
+    req_state* s,
+    const ceph::real_time &obj_mtime) {
+  auto attr_iter = attrs.find(RGW_ATTR_OBJ_REPLICATION_STATUS);
+  if (attr_iter != attrs.end() && attr_iter->second.to_str() == "PENDING") {
+    if (s->object->is_sync_completed(dpp, obj_mtime)) {
+        s->object->set_atomic();
+        rgw::sal::Attrs attrs, rmattrs;
```

Comment:
this `attrs` variable shadows the function parameter, please rename one

when we update RGW_ATTR_OBJ_REPLICATION_STATUS to "COMPLETED" here, does that updated value get sent in the x-amz-replication-status header?

Reply:
renamed the local `attrs` variable. Yes, this updated value is reflected in the header.